File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        int minLen=INT_MAX;;
        for(int i=0;i<strs.size();i++){
            if(strs[i].length()<minLen)minLen=strs[i].length();
        }
        
        bool flag=false;
        for(int j=0;j<minLen;j++){
            char ch=strs[0][j];
            for(int i=1;i<strs.size();i++){
                if(ch!=strs[i][j])flag=true;
            }
            if(flag)break;
            ans+=ch;
        }
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String ans="";

        int mini=Integer.MAX_VALUE;
        for(int i=0;i<strs.length;i++){
            int len=strs[i].length();
           mini= Math.min(len,mini);
        }

        for(int j=0;j<mini;j++){
            char ch=strs[0].charAt(j);
            boolean flag=false;    
            for(int i=1;i<strs.length;i++){
                if(ch!=strs[i].charAt(j))flag=true;
            }
            if(flag)break;
            else ans+=ch;
        }
        return ans;
    }
}